cli/dstack/backend/base/__init__.py

```diff
@@ -240,6 +243,37 @@ def get_configurator(cls) -> Configurator:
     def delete_workflow_cache(self, workflow_name: str):
         pass
 
+    def run_workflow(
+        self,
+        workflow_name: str,
+        tag_name: Optional[str] = None,
+        ssh_pub_key: Optional[str] = None,
+        args: Optional[argparse.Namespace] = None,
+    ) -> (str, List[Job]):
+        """Runs workflow or provider by name
+        :return: run_name, jobs
+        """
+        workflow = self.repo.get_workflows(credentials=self.get_repo_credentials()).get(
+            workflow_name, {}
+        )
+        if workflow:
+            provider = providers.load_provider(workflow["provider"])
+        elif workflow_name in providers.get_provider_names():
+            provider = providers.load_provider(workflow_name)
+            workflow_name = None
+        else:
+            raise NameError(f"No workflow or provider '{workflow_name}' is found")
```

Comment:
Errors are part of the API, and we should define our own errors. `NameError` has specific different semantics.